### PR TITLE
Initial updates for supporting s3 bucket logging targets.

### PR DESCRIPTION
**Pull request by jgadling**
### Summary
Add support for logging S3 access to another bucket or prefix to the private bucket module.

### Test Plan
I generated a tf file that created two buckets (a content bucket, and a logging bucket) and set the logging config of one bucket to refer to the other. The logging config in the plan had logging enabled for the content bucket and disabled for the logging bucket.

Tests pass.

### References
(Optional) Additional links to provide more context.

### CODE REVIEW

File: aws-s3-private-bucket/main.tf [hunk 1]
```diff
@@ -110,11 +110,13 @@ resource "aws_s3_bucket" "bucket" {
...
-  #   target_bucket = ""
-  #   target_prefix = ""
-  # }
+  dynamic "logging" {
+    for_each = var.logging_bucket == null ? [] : [var.logging_bucket]
+    content {
+      target_bucket = var.logging_bucket.name
+      target_prefix = var.logging_bucket.prefix
+    }
+  }
```
Comment (edulop91) [inline aws-s3-private-bucket/main.tf:119]: do you know what's the difference between this and cloudtrail logs?

File: aws-s3-private-bucket/main.tf [hunk 2]
```diff
@@ -110,11 +110,13 @@ resource "aws_s3_bucket" "bucket" {
...
 
-  # TODO
-  #   logging {
-  #   target_bucket = ""
-  #   target_prefix = ""
-  # }
+  dynamic "logging" {
+    for_each = var.logging_bucket == null ? [] : [var.logging_bucket]
+    content {
+      target_bucket = var.logging_bucket.name
```
Comment (mbarrien) [inline aws-s3-private-bucket/main.tf:116]: Don't know if it's better to do each.name/each.prefix, or leave it like this.

File: aws-s3-private-bucket/main.tf (hunk 1 above)
Comment (jgadling) [reply aws-s3-private-bucket/main.tf:119]: Here's a pretty interesting comparison: https://www.netskope.com/blog/aws-s3-logjam-server-access-logging-vs-object-level-logging

It looks like cloudtrail is more complete and more reliable, but also more complicated to configure.

Comment (maniarathi) [reply aws-s3-private-bucket/main.tf:119]: There's a table in here that I used to determine that S3 logs was the way forward: https://docs.aws.amazon.com/AmazonS3/latest/userguide/logging-with-S3.html. Ultimately it was a combination of ease to set up (took me just a few minutes to set up and start querying) plus the fact that the S3 logs contain more useful information for us (i.e. Object Size, Total Time, Turn-Around Time, and HTTP Referer).

**Review by maniarathi — APPROVED** [review]

File: aws-s3-private-bucket/main.tf (hunk 2 above)
Comment (jgadling) [reply aws-s3-private-bucket/main.tf:116]: yeah, I could go either way as well. I think the each syntax is a little bit more confusing though for an iterator where we'll always have 0 or 1 loops.

**Review by mbarrien — APPROVED** [review]

**Pull request merged**